src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfTranscriptFeature.java

```diff
@@ -0,0 +1,144 @@
+                                           String transcriptId,
+                                           GeneTranscriptType geneType,
+                                           GeneTranscriptStatus geneStatus,
+                                           String geneName,
+                                           GeneTranscriptType transcriptType,
+                                           GeneTranscriptStatus transcriptStatus,
+                                           String transcriptName,
+                                           int exonNumber,
+                                           String exonId,
+                                           LocusLevel locusLevel,
+                                           ArrayList<OptionalField<?>> optionalFields,
+                                           String anonymousOptionalFields) {
+
+        return new GencodeGtfTranscriptFeature(featureOrderNumber, chromosomeName, annotationSource, featureType, genomicStartLocation, genomicEndLocation, genomicStrand, genomicPhase, geneId, transcriptId, geneType, geneStatus, geneName, transcriptType, transcriptStatus, transcriptName, exonNumber, exonId, locusLevel, optionalFields, anonymousOptionalFields);
+    }
+
+    // ================================================================================================
+
+    private ArrayList<GencodeGtfExonFeature>            exons = new ArrayList<>();
+    private ArrayList<GencodeGtfSelenocysteineFeature>  selenocysteines = new ArrayList<>();
+    private ArrayList<GencodeGtfUTRFeature>             utrs = new ArrayList<>();
+
+    // ================================================================================================
+
+    public ArrayList<GencodeGtfExonFeature> getExons() {
+        return exons;
+    }
+
+    public void addExon( GencodeGtfExonFeature exon ) {
+        exons.add(exon);
+    }
+
+    public ArrayList<GencodeGtfSelenocysteineFeature> getSelenocysteines() {
+        return selenocysteines;
+    }
+
+    public void addSelenocysteine( GencodeGtfSelenocysteineFeature selenocysteine ) {
+        selenocysteines.add(selenocysteine);
+    }
+
+    public ArrayList<GencodeGtfUTRFeature> getUtrs() {
+        return utrs;
+    }
+
+    public void addUtr( GencodeGtfUTRFeature utr ) { utrs.add(utr); }
+
+    @Override
+    protected List<GencodeGtfFeature> getAllFeatures() {
+        ArrayList<GencodeGtfFeature> list = new ArrayList<>();
+        list.add(this);
+
+        for ( GencodeGtfExonFeature feature : exons ) {
+            list.addAll(feature.getAllFeatures());
+        }
+
+        for ( GencodeGtfSelenocysteineFeature feature : selenocysteines ) {
+            list.addAll(feature.getAllFeatures());
+        }
+
+        for ( GencodeGtfUTRFeature feature : utrs ) {
+            list.addAll(feature.getAllFeatures());
+        }
+
+        return list;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Same comment here about overriding `hashCode()` and ensuring consistency with `equals()`